programs/{{project-name}}/src/lib.rs

```diff
@@ -46,19 +46,20 @@ pub mod {{rust-name}} {
         let pool_type = [0u8; 32];
         let mut program_id_hash = hash(&ctx.program_id.to_bytes()).to_bytes();
         program_id_hash[0] = 0;
-        let checked_inputs = [
+        let mut checked_inputs = [
             program_id_hash,
             inputs_des.transaction_hash,
-            // inputs_des.current_slot.to_vec(),
-            [0u8; 32],
         ];
+        for _ in 2..NR_CHECKED_INPUTS {
+            checked_inputs.push([0u8; 32]);
+        }
         process_psp_instruction_first::<NR_CHECKED_INPUTS, 17>(
             ctx,
             &proof,
             &public_amount,
             &inputs_des.input_nullifier,
             &inputs_des.output_commitment,
-            &checked_inputs,
+            &checked_inputs.try_into().unwrap(),
```

Comment:
And this way, I think we could get rid of `.try_info().unwrap()` here, we could just do `&checked_inputs`.

Reply:
the issue is that checked_inputs expects a fixed sized array of length public inputs but to make it variable sergey is using a vector above

Reply:
the reason why we need to change it in the first place is that right now it's hard coded to 3 public inputs for a psp, if you try to add more you run into the error in the pr description.

Reply:
I don't get your point. You agreed with my comment above (defining `checked_inputs` as an array of `NR_CHECKED_INPUTS` size). After doing it, we can just pass a slice to it (`checked_inputs` is an array, `&checked_inputs` is a slice). We don't need to convert any vectors anymore. And the size depends on the const generic.

If `NR_CHECKED_INPUTS` is 3, we are still fine, because above we would allocate an array with size 3, put two values and have one zero there.

Reply:
If we would like to put more inputs than 3, sure, we have a problem, but totally unrelated to the `.try_into().unwrap()` here? That would require a bigger change and bumping the size on the Light Protocol side, if I'm not mistaken?